### ISSUE TITLE
export

### ISSUE THREAD

**Issue by alicera** [ultralytics/yolov5#3396]
## ❔Question

PyTorch: starting from runs/train/night_m_0531/weights/last.pt (42.5 MB)
ONNX: starting export with onnx 1.8.0...
/work/Pictures/YOLOv5/new_yolov5_0515/yolov5/models/yolo.py:51: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
ONNX: simplifying with onnx-simplifier 0.3.5...
ONNX: export success, saved as runs/train/night_m_0531/weights/last.onnx (84.3 MB)

**Comment by glenn-jocher** [comment]
@alicera you can skip grid ops during export by placing model in train mode:
```
python export.py --train
```

though you can also ignore these messages, they are warnings not errors.

**Comment by alicera** [comment]
parser.add_argument('--train', action='store_true', help='model.train() mode')

If I set the --train , it will be model.train().
Why not model.eval() for inference?

**Comment by alicera** [comment]
I use --train, it still

PyTorch: starting from runs/train/weights/last.pt (42.5 MB)
ONNX: starting export with onnx 1.8.0...
/work/Pictures/YOLOv5/yolov5/models/yolo.py:51: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:

**Comment by glenn-jocher** [comment]
@alicera your code is out of date. To update your code:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
